src/net.cpp

```diff
@@ -86,7 +86,8 @@ std::map<CNetAddr, LocalServiceInfo> mapLocalHost GUARDED_BY(cs_mapLocalHost);
 static bool vfLimited[NET_MAX] GUARDED_BY(cs_mapLocalHost) = {};
 std::string strSubVersion;
 
-limitedmap<uint256, int64_t> mapAlreadyAskedFor(MAX_INV_SZ);
+limitedmap<uint256, int64_t> tx_to_ask(MAX_INV_SZ);
```

Comment:
If an adversary were to blast us with huge INV messages, then I think they could take over this data structure, and blind us to all other transactions -- once a txid leaves `tx_to_ask` I think we'll never request it?

Also, I think this is one way that `tx_to_ask` and `tx_to_ask_from` could get out of sync which looks like it would cause a memory leak.